libs/core/libs/shared/src/api.rs

```diff
@@ -93,6 +96,7 @@ pub enum ChangeDocError {
     DocumentNotFound,
     DocumentDeleted,
     NotPermissioned,
+    FileNameTooLong,
```

Comment:
> @parth: prob doesn't need to be here
the file name length should be checked both in upsert and when the user renames. So it should be included.